sklearn/manifold/_classical_mds.py

```diff
@@ -0,0 +1,182 @@
+       Groenen P. Springer Series in Statistics (1997)
+
+    Examples
+    --------
+    >>> from sklearn.datasets import load_digits
+    >>> from sklearn.manifold import ClassicalMDS
+    >>> X, _ = load_digits(return_X_y=True)
+    >>> X.shape
+    (1797, 64)
+    >>> cmds = ClassicalMDS(n_components=2)
+    >>> X_emb = cmds.fit_transform(X[:100])
+    >>> X_emb.shape
+    (100, 2)
+    """
+
+    _parameter_constraints: dict = {
+        "n_components": [Interval(Integral, 1, None, closed="left")],
+        "dissimilarity": [str, callable],
+    }
+
+    def __init__(
+        self,
+        n_components=2,
+        *,
+        dissimilarity="euclidean",
+    ):
+        self.n_components = n_components
+        self.dissimilarity = dissimilarity
+
+    def __sklearn_tags__(self):
+        tags = super().__sklearn_tags__()
+        tags.input_tags.pairwise = self.dissimilarity == "precomputed"
+        return tags
+
+    def fit(self, X, y=None):
+        """
+        Compute the embedding positions.
+
+        Parameters
+        ----------
+        X : array-like of shape (n_samples, n_features) or \
+                (n_samples, n_samples)
+            Input data. If ``dissimilarity=='precomputed'``, the input should
+            be the dissimilarity matrix.
+
+        y : Ignored
+            Not used, present for API consistency by convention.
+
+        Returns
+        -------
+        self : object
+            Fitted estimator.
+        """
+        self.fit_transform(X)
+        return self
+
+    @_fit_context(prefer_skip_nested_validation=True)
+    def fit_transform(self, X, y=None):
+        """
+        Compute the embedding positions.
+
+        Parameters
+        ----------
+        X : array-like of shape (n_samples, n_features) or \
+                (n_samples, n_samples)
+            Input data. If ``dissimilarity=='precomputed'``, the input should
+            be the dissimilarity matrix.
+
+        y : Ignored
+            Not used, present for API consistency by convention.
+
+        Returns
+        -------
+        X_new : ndarray of shape (n_samples, n_components)
+            The embedding coordinates.
+        """
+
+        X = validate_data(self, X)
+
+        if self.dissimilarity == "precomputed":
+            self.dissimilarity_matrix_ = X
+            self.dissimilarity_matrix_ = check_symmetric(
+                self.dissimilarity_matrix_, raise_exception=True
+            )
+        else:
+            self.dissimilarity_matrix_ = pairwise_distances(
+                X, metric=self.dissimilarity
+            )
+
+        # Double centering
+        B = self.dissimilarity_matrix_**2
+        B = B.astype(np.float64)
+        B -= np.mean(B, axis=0)
+        B -= np.mean(B, axis=1, keepdims=True)
+        B *= -0.5
+
+        # Eigendecomposition
+        w, U = linalg.eigh(B)
+        w = w[::-1][: self.n_components]
+        U = U[:, ::-1][:, : self.n_components]
```

Comment:
I guess `eigh` returns the eigenvalues in increasing order while we need them in decreasing order ?
Could you add a small comment on reversing the arrays ?

Reply:
Added.